src/components/SharedComponents/TaxonResult.tsx

```diff
@@ -192,51 +230,52 @@ const TaxonResult = ( {
             </View>
           )}
         </View>
-      </Pressable>
-      <View className="flex-row items-center">
-        { !hideInfoButton && (
-          <INatIconButton
-            icon="info-circle-outline"
-            size={22}
-            onPress={( ) => {
-              if ( typeof ( onPressInfo ) === "function" ) {
-                onPressInfo( usableTaxon );
-                return;
-              }
-              navToTaxonDetails( );
-            }}
-            color={String(
-              clearBackground
-                ? colors.white
-                : colors.darkGray
-            )}
-            accessibilityLabel={t( "More-info" )}
-            accessibilityHint={t( "Navigates-to-taxon-details" )}
-          />
-        )}
-        { showCheckmark
-          && renderCheckmark()}
-        { showEditButton
+      </TaxonResultMain>
+      { !unpressable && (
+        <View className="flex-row items-center">
+          { !hideInfoButton && (
+            <INatIconButton
+              icon="info-circle-outline"
+              size={22}
+              onPress={( ) => {
+                if ( typeof ( onPressInfo ) === "function" ) {
+                  onPressInfo( usableTaxon );
+                  return;
+                }
+                navToTaxonDetails( );
+              }}
+              color={String(
+                clearBackground
+                  ? colors?.white
+                  : colors?.darkGray
+              )}
+              accessibilityLabel={t( "More-info" )}
+              accessibilityHint={t( "Navigates-to-taxon-details" )}
+            />
+          )}
+          { showCheckmark && renderCheckmark()}
+          { showEditButton && handleTaxonOrEditPress
+              && (
+                <INatIconButton
+                  icon="edit"
+                  size={20}
+                  onPress={handleTaxonOrEditPress}
+                  accessibilityLabel={t( "Edit-identification" )}
+                  accessibilityHint={t( "Edits-this-observations-taxon" )}
+                />
+              )}
+          { showRemoveButton && handleRemovePress
             && (
               <INatIconButton
-                icon="edit"
+                icon="close"
```

Comment:
These lines highlight an avoidable difficulty caused by nesting.

Reply:
What is the difficulty, exactly? Do just mean you've got some junky lines in the diff due to changes in indentation?

Reply:
Yes. 
![CleanShot 2024-12-11 at 17 11 53@2x](https://github.com/user-attachments/assets/ec9c41f0-d328-45a3-9033-2efbcf0e765b)